controllers/testdata_test.go

```diff
@@ -0,0 +1,146 @@
+package controllers
```

Comment:
Interesting concept of putting all data into one file.

I, for one, like to see data and test referencing each other - yes with duplication. In my book, tests work a bit different than production code. But we shall leave this as-is.